File: Ch7_FilterDesignTechniques/Middle_WindowEffect/test_butterworth.m
FS = 30;

LPF = 0.5;
HPF = 3.3;

WinSec=1.6; %(was a 32 frame window with 20 fps camera)

NyquistF = 1/2*FS;
[B,A] = butter(3,[LPF/NyquistF HPF/NyquistF]);
imp_rep = filter(B,A,double(imp));

% Zero padding to observe. Uncomment it and think.
imp_rep =[zeros(1,512) imp_rep zeros(1,512)];

title="butter filter 3 order";
filter_analysis_imp(imp_rep,fps,title);

File: Ch7_FilterDesignTechniques/Middle_WindowEffect/test_butterworth.py
import numpy as np
from scipy.signal import butter, lfilter
from filter_analysis_imp import filter_analysis_imp

FS = 30
LPF = 0.5
HPF = 3.3
WinSec = 1.6  # unused but kept for reference

NyquistF = 0.5 * FS
B, A = butter(3, [LPF / NyquistF, HPF / NyquistF], btype="band")

n = np.arange(-511, 513)
imp = (n == 0).astype(float)

imp_rep = lfilter(B, A, imp)
imp_rep = np.concatenate([np.zeros(512), imp_rep, np.zeros(512)])

filter_analysis_imp(imp_rep, FS, "butter filter 3 order")
